Celeste.Mod.mm/Mod/UI/TextMenuExt.cs

```diff
@@ -238,5 +238,128 @@ public EaseInSubHeaderExt(string title, bool initiallyVisible, TextMenu containi
+                    if (Index == 0) {
+                        Audio.Play("event:/ui/main/button_toggle_on");
+                    } else {
+                        Audio.Play("event:/ui/main/button_toggle_off");
+                    }
+                    PreviousIndex = Index;
+                    Index = 1 - Index;
+                    lastDir = ((Index == 1) ? 1 : -1);
+                    ValueWiggler.Start();
+                    if (OnValueChange != null) {
+                        OnValueChange(Index);
+                    }
+                }
+            }
+
+            public override void Update() {
+                sine += Engine.RawDeltaTime;
+            }
+            public override float LeftWidth() {
+                return ActiveFont.Measure(Label).X + 32f;
+            }
+            public override float RightWidth() {
+                float num = 0f;
+                num = Math.Max(num, ActiveFont.Measure(min.ToString()).X);
+                num = Math.Max(num, ActiveFont.Measure(max.ToString()).X);
+                return num + 120f;
+            }
+            public override float Height() {
+                return ActiveFont.LineHeight;
+            }
+
+            public override void Render(Vector2 position, bool highlighted) {
+                float alpha = Container.Alpha;
+                Color strokeColor = Color.Black * (alpha * alpha * alpha);
+                Color color = Disabled ? Color.DarkSlateGray : ((highlighted ? Container.HighlightColor : Color.White) * alpha);
+                ActiveFont.DrawOutline(Label, position, new Vector2(0f, 0.5f), Vector2.One, color, 2f, strokeColor);
+                if ((max - min) > 0) {
+                    float num = RightWidth();
+                    ActiveFont.DrawOutline(Index.ToString(), position + new Vector2(Container.Width - num * 0.5f + (float) lastDir * ValueWiggler.Value * 8f, 0f), new Vector2(0.5f, 0.5f), Vector2.One * 0.8f, color, 2f, strokeColor);
+                    Vector2 vector = Vector2.UnitX * (highlighted ? ((float) Math.Sin((double) (sine * 4f)) * 4f) : 0f);
+                    bool flag = Index > min;
+                    Color color2 = flag ? color : (Color.DarkSlateGray * alpha);
+                    Vector2 position2 = position + new Vector2(Container.Width - num + 40f + ((lastDir < 0) ? (-ValueWiggler.Value * 8f) : 0f), 0f) - (flag ? vector : Vector2.Zero);
+                    ActiveFont.DrawOutline("<", position2, new Vector2(0.5f, 0.5f), Vector2.One, color2, 2f, strokeColor);
+                    bool flag2 = Index < max;
+                    Color color3 = flag2 ? color : (Color.DarkSlateGray * alpha);
+                    Vector2 position3 = position + new Vector2(Container.Width - 40f + ((lastDir > 0) ? (ValueWiggler.Value * 8f) : 0f), 0f) + (flag2 ? vector : Vector2.Zero);
+                    ActiveFont.DrawOutline(">", position3, new Vector2(0.5f, 0.5f), Vector2.One, color3, 2f, strokeColor);
```

Comment:
Certain local variables can be reused instead of defining new ones (`color2`, `position3`, ...), the flag variables are unnecessary (`bool flag2 = cond; ... flag2 ? x : y` vs `... cond ? x : y`), some casts seem to be compiler clutter (`(float) lastDir * anotherFloatValue`, `(double) (sine * 4f)`).